ecosystem/typescript/sdk_v2/src/crypto/ed25519.ts

```diff
@@ -0,0 +1,212 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import nacl from "tweetnacl";
+import { Deserializer } from "../bcs/deserializer";
+import { Serializer } from "../bcs/serializer";
+import { Hex } from "../core/hex";
+import { HexInput } from "../types";
+import * as asymmetric_crypto from "./asymmetric_crypto";
+
+export class PublicKey extends asymmetric_crypto.PublicKey {
```

Comment:
this is a weird js usage. I assume you had naming issue by doing `import {PublicKey, PrivateKey, Signature} from "./asymmetric_crypto";` . if we have an abstract class for each, lets change the derived class name then?
```
export class Ed25519PublicKey
```

Reply:
I was trying to mimic closely with Python SDK lol

But yeah, `Ed25519PublicKey` works